packages/contracts-bedrock/scripts/Artifacts.s.sol

```diff
@@ -59,17 +59,6 @@ abstract contract Artifacts {
         try vm.createDir(deploymentsDir, true) { } catch (bytes memory) { }
 
         uint256 chainId = vm.envOr("CHAIN_ID", block.chainid);
-        string memory chainIdPath = string.concat(deploymentsDir, string("/.chainId"));
-        try vm.readFile(chainIdPath) returns (string memory localChainId) {
-            if (vm.envOr("STRICT_DEPLOYMENT", true)) {
-                require(
-                    vm.parseUint(localChainId) == chainId,
-                    string.concat("Misconfigured networks: ", localChainId, " != ", vm.toString(chainId))
-                );
-            }
-        } catch {
-            vm.writeFile(chainIdPath, vm.toString(chainId));
-        }
```

Comment:
Is there any way we can keep this check for non-foundry tests (ie when running the `forge script` subcommand)? If not I think it should be ok

Reply:
I think the check should not be based on a file. If we want the check then it could come from an env var. Its very opinionated to force an env var but adds max safety. Security isn't asking for that feature so I am not sure if its necessary. Def open to asking them and following up with an addition if its something that they would use